app/util/legUtils.js

```
@@ -865,3 +865,27 @@ export const legDestination = (intl, leg, secondary, nextLeg = null) => {
   }
   return intl.formatMessage({ id, defaultMessage: 'place' });
 };
+
+export const isPlatformChanged = leg => {
+  if (!leg?.trip || !leg.start?.scheduledTime) {
```

Would it be  better to remove ? after leg and let the code crash if this is called with undefined leg? Is it possible that leg has no start?

---

I'm more comfortable with the leg having the ?, because of the intricaties of the travel companion legs, but there really shouldn't be a leg with no start.